### PR TITLE
Add missing `@types` for used libraries

### PR DESCRIPTION
**Pull request by Tristan-WorkGH**
Add missing `@types` for used libraries, needed to develop correctly in Typescript.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,8 +69,27 @@
...
         "@babel/preset-react": "^7.24.1",
         "@babel/preset-typescript": "^7.24.1",
-        "@types/react": "^18.2.73",
-        "@types/react-dom": "^18.2.23",
+        "@types/core-js": "^2.5.8",
+        "@types/dagre": "^0.7.52",
+        "@types/eslint": "^7.29.0",
+        "@types/eslint-config-prettier": "^6.11.3",
+        "@types/identity-obj-proxy": "^3.0.2",
+        "@types/jest": "^29.5.12",
```
Comment (TheMaskedTurtle) [inline package.json:77]: built-in

File: package.json [hunk 2]
```diff
@@ -73,8 +69,27 @@
...
+        "@types/license-checker": "^25.0.6",
+        "@types/localized-countries": "^2.0.3",
+        "@types/lucene-escape-query": "^1.0.2",
+        "@types/node": "^18.19.29",
```
Comment (TheMaskedTurtle) [inline package.json:81]: Same

Comment (Tristan-WorkGH) [reply package.json:81]: I'm not sure for this one. Can you give more details?

Comment (TheMaskedTurtle) [reply package.json]: alright

**Pull request merged**